Remove 'iAd Producer' quirk

karlcow opened the pull request:
#### 7370a17dd48726368e9940cc4c9584f208c20362
<pre>
Remove &apos;iAd Producer&apos; quirk
<a href="https://bugs.webkit.org/show_bug.cgi?id=268000">https://bugs.webkit.org/show_bug.cgi?id=268000</a>
<a href="https://rdar.apple.com/122010865">rdar://122010865</a>

Reviewed by Ryosuke Niwa and Anne van Kesteren.

This removes an old Quirk which was put in place a long time ago.
The product was discontinued in 2016.

* Source/WebCore/dom/MouseEvent.cpp:
(WebCore::MouseEvent::initMouseEventQuirk): Deleted.
* Source/WebCore/dom/MouseEvent.h:
* Source/WebCore/dom/MouseEvent.idl:
* Source/WebCore/platform/RuntimeApplicationChecks.h:
* Source/WebCore/platform/cocoa/RuntimeApplicationChecksCocoa.mm:
(WebCore::MacApplication::isIAdProducer): Deleted.

Canonical link: <a href="https://commits.webkit.org/274282@main">https://commits.webkit.org/274282@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f9f8c23988d390d17c4482581e6e392190814f99

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38496 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17436 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40833 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41043 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/34187 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/40801 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/20217 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/14781 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/32383 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39069 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14688 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33497 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12785 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12742 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/34357 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42319 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34969 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34756 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38580 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13350 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11038 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36784 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14950 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8652 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13817 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/14422 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders commented on Source/WebCore/dom/MouseEvent.idl:
```diff
@@ -44,13 +44,11 @@
-    // FIXME: relatedTarget should be of type EventTarget? but we need to use any to support a quirk for iAd Producer (rdar://problem/30640101).
-    [CallWith=CurrentGlobalObject&CurrentScriptExecutionContext, ImplementedAs=initMouseEventQuirk] undefined initMouseEvent([AtomString] DOMString type,
-        optional boolean canBubble = false, optional boolean cancelable = false,
+    initMouseEvent(optional DOMString type = "undefined", optional boolean canBubble = false, optional boolean cancelable = false,
```
WebIDL requires you specify the return type of the function, so:
```suggestion
    undefined initMouseEvent(optional DOMString type = "undefined", optional boolean canBubble = false, optional boolean cancelable = false,
```

Maybe worth filing a bug on the fact that the IDL generation didn't fail the build? This should be a parse error, I think.

karlcow replied:
Thanks @gsnedders 
Trying something

Ahmad-S792 replied:
> WebIDL requires you specify the return type of the function, so:
> 
> Maybe worth filing a bug on the fact that the IDL generation didn't fail the build? This should be a parse error, I think.

Hi Sam, this is second time where IDL generation didn't failed. I had similar instances in TextTrack / AudioTrack (I can find relevant commit), where we didn't added `readonly` but it compiled and when added, we progressed few WPT. So I think it is worth raising bug.

karlcow replied:
https://bugs.webkit.org/show_bug.cgi?id=268966
@gsnedders feel free to make it clearer.

annevk commented on Source/WebCore/dom/MouseEvent.idl:
```diff
@@ -44,13 +44,12 @@
-    [CallWith=CurrentGlobalObject&CurrentScriptExecutionContext, ImplementedAs=initMouseEventQuirk] undefined initMouseEvent([AtomString] DOMString type,
-        optional boolean canBubble = false, optional boolean cancelable = false,
+    // FIXME: Using "undefined" as default parameter value is wrong.
+    undefined initMouseEvent(optional [AtomString] DOMString type = "undefined", optional boolean canBubble = false, optional boolean cancelable = false,
```
I don't really understand why `type` is changed from a required argument to an optional argument defaulting to "`undefined`". This is also not explained by the commit message.

Ahmad-S792 replied:
Web-Spec - https://w3c.github.io/uievents/#idl-interface-MouseEvent-initializers ; we should just sync this.

gsnedders replied:
Notwithstanding aligning with the spec is good, it probably makes more sense to remove the quirk separate to making other WebIDL changes.

karlcow replied:
yes it's what I was trying to do initially, without success. 
Just "reverting" is not working. 
Note that this is the initial code before the Quirk…
https://github.com/WebKit/WebKit/commit/51b46788f5aa0519a2c829cd6a01569e468ff568

annevk replied:
But at some later point that was tightened up, no? Because you are changing this away from `[AtomString] DOMString type` after all.

What exactly is not working?

annevk commented on Source/WebCore/dom/MouseEvent.idl:
```diff
@@ -44,13 +44,12 @@
-    [CallWith=CurrentGlobalObject&CurrentScriptExecutionContext, ImplementedAs=initMouseEventQuirk] undefined initMouseEvent([AtomString] DOMString type,
-        optional boolean canBubble = false, optional boolean cancelable = false,
+    // FIXME: Using "undefined" as default parameter value is wrong.
+    undefined initMouseEvent(optional [AtomString] DOMString type, optional boolean canBubble = false, optional boolean cancelable = false,
```
This is still wrong. `type` shouldn't be optional. That's also why you regressed WPT tests. (The FIXME is also wrong.)

karlcow replied:
🤦 Don't code quickly when you are sick with headaches. 
Thanks @annevk

karlcow committed:
Remove 'iAd Producer' quirk
https://bugs.webkit.org/show_bug.cgi?id=268000
rdar://122010865

Reviewed by Ryosuke Niwa and Anne van Kesteren.

This removes an old Quirk which was put in place a long time ago.
The product was discontinued in 2016.

* Source/WebCore/dom/MouseEvent.cpp:
(WebCore::MouseEvent::initMouseEventQuirk): Deleted.
* Source/WebCore/dom/MouseEvent.h:
* Source/WebCore/dom/MouseEvent.idl:
* Source/WebCore/platform/RuntimeApplicationChecks.h:
* Source/WebCore/platform/cocoa/RuntimeApplicationChecksCocoa.mm:
(WebCore::MacApplication::isIAdProducer): Deleted.

Canonical link: https://commits.webkit.org/274282@main
